tests/components/recorder/test_purge.py

```diff
@@ -13,226 +12,216 @@
+        assert not finished
+        assert states.count() == 4
+
+        finished = purge_old_data(hass.data[DATA_INSTANCE], 4, repack=False)
+        assert not finished
+        assert states.count() == 2
+
+        finished = purge_old_data(hass.data[DATA_INSTANCE], 4, repack=False)
+        assert finished
+        assert states.count() == 2
+
+
+def test_purge_old_events(hass, hass_recorder):
+    """Test deleting old events."""
+    hass = hass_recorder()
+    _add_test_events(hass)
+
+    with session_scope(hass=hass) as session:
+        events = session.query(Events).filter(Events.event_type.like("EVENT_TEST%"))
+        assert events.count() == 6
+
+        # run purge_old_data()
+        finished = purge_old_data(hass.data[DATA_INSTANCE], 4, repack=False)
+        assert not finished
+        assert events.count() == 4
+
+        finished = purge_old_data(hass.data[DATA_INSTANCE], 4, repack=False)
+        assert not finished
+        assert events.count() == 2
+
+        # we should only have 2 events left
+        finished = purge_old_data(hass.data[DATA_INSTANCE], 4, repack=False)
+        assert finished
+        assert events.count() == 2
+
+
+def test_purge_method(hass, hass_recorder):
+    """Test purge method."""
+    hass = hass_recorder()
+    service_data = {"keep_days": 4}
+    _add_test_events(hass)
+    _add_test_states(hass)
+    _add_test_recorder_runs(hass)
+
+    # make sure we start with 6 states
+    with session_scope(hass=hass) as session:
+        states = session.query(States)
+        assert states.count() == 6
+
+        events = session.query(Events).filter(Events.event_type.like("EVENT_TEST%"))
+        assert events.count() == 6
+
+        recorder_runs = session.query(RecorderRuns)
+        assert recorder_runs.count() == 7
+
+        hass.data[DATA_INSTANCE].block_till_done()
+        wait_recording_done(hass)
+
+        # run purge method - no service data, use defaults
+        hass.services.call("recorder", "purge")
+        hass.block_till_done()
+
+        # Small wait for recorder thread
+        hass.data[DATA_INSTANCE].block_till_done()
+        wait_recording_done(hass)
+
+        # only purged old events
+        assert states.count() == 4
+        assert events.count() == 4
+
+        # run purge method - correct service data
+        hass.services.call("recorder", "purge", service_data=service_data)
+        hass.block_till_done()
+
+        # Small wait for recorder thread
+        hass.data[DATA_INSTANCE].block_till_done()
+        wait_recording_done(hass)
+
+        # we should only have 2 states left after purging
+        assert states.count() == 2
+
+        # now we should only have 2 events left
+        assert events.count() == 2
+
+        # now we should only have 3 recorder runs left
+        assert recorder_runs.count() == 3
+
+        assert not ("EVENT_TEST_PURGE" in (event.event_type for event in events.all()))
```

Comment:
Codacy found an issue: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=6294088)